### PR TITLE
Feat/sql parsing

### PR DESCRIPTION
**Pull request by svaningelgem**

### CODE REVIEW

File: pysparkling/sql/ast/tests/test_functions.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from pysparkling.sql.types import IntegerType, StructField, StructType
+
+ROW = Row(a=1, b=2, c=3)
+SCHEMA = StructType([
+    StructField("a", IntegerType()),
+    StructField("b", IntegerType()),
+])
+
+
+SCENARIOS = {
```
Comment (ramoreira) [inline pysparkling/sql/ast/tests/test_functions.py:14]: Are all functions (currently) supported functions being covered?

File: pysparkling/sql/ast/tests/test_type_parsing.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+from pysparkling import Context
+from pysparkling.sql.ast.ast_to_python import parse_data_type, SqlParsingError
+from pysparkling.sql.session import SparkSession
+from pysparkling.sql.types import (
+    ArrayType, BinaryType, BooleanType, ByteType, DateType, DecimalType, DoubleType, FloatType, IntegerType, LongType,
+    MapType, NullType, ShortType, StringType, StructField, StructType, TimestampType
+)
+
+DATA_TYPE_SCENARIOS = {
```
Comment (ramoreira) [inline pysparkling/sql/ast/tests/test_type_parsing.py:14]: Quite an extensive list of type scenarios, great!

**Comment by ramoreira** [comment]
The SQL parsing feature is definitely going in the right direction.

Idea for future PRs: start documenting supported/unsupported features.

**Review by ramoreira — APPROVED** [review]

File: pysparkling/sql/ast/tests/test_functions.py (hunk 1 above)
Comment (svaningelgem) [reply pysparkling/sql/ast/tests/test_functions.py:14]: this is only for the datatypes. I didn't verify the functions yet.

**Pull request merged**